cdk/dcm_floodlight/transform.js

```diff
@@ -57,7 +60,34 @@ function trackPostMapper(input, mappedPayload, rudderContext) {
 
   const baseEndpoint = "https://ad.doubleclick.net/ddm/activity/";
 
-  let event = getValueFromMessage(message, "event");
+  let event;
+  if (message.type === "page") {
+    const { category } = message.properties;
+    const { name } = message || message.properties;
+
+    if (!category && !name) {
+      throw new CustomError(
+        "[DCM Floodlight]:: category or name is required for page",
+        400
+      );
+    }
+
+    if (category && name) {
+      message.event = `Viewed ${category} ${name} Page`;
+      message.type = "track";
+    } else if (category) {
+      // categorized pages
+      message.event = `Viewed ${category} Page`;
+      message.type = "track";
+    } else {
+      // named pages
+      message.event = `Viewed ${name} Page`;
+      message.type = "track";
+    }
+  }
+
+  event = getValueFromMessage(message, "event");
```

Comment:
We can simply access `message.event` directly right ?!
Is there is a specific reason to why we are using `getValueFromMessage` function here ?

Reply:
Since it's a 1st level child taking it directly should be fine. Incorporated the changes